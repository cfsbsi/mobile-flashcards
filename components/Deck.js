import React from 'react'
import {View, Text} from 'react-native'
import styled from 'styled-components/native';


export default function Deck(props) {
    const {quantity, title} = props;
    const cardQuantitiesLiteral = `${quantity} ${quantity===1?'card':'cards'}`;

    return (
        <DeckStyled>
            <Title onPress={() => props.navigation.navigate('SingleViewDeck', {title: title, subtitle: cardQuantitiesLiteral})} >{title}</Title>
            <Text>{cardQuantitiesLiteral}</Text>
        </DeckStyled>
    )
}

const DeckStyled = styled.View`
    align-items: center;
    padding-top: 20;
    padding-bottom: 20;
    border-bottom-color: black;
    border-bottom-width: 2;
`

const Title = styled.Text`
    font-size: 30;
    font-weight: bold;
`

