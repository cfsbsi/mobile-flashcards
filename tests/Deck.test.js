import React from 'react';
import Deck from '../components/Deck';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
    const tree = renderer.create(<Deck quantity={1}/>).toJSON();
    expect(tree).toMatchSnapshot();
});