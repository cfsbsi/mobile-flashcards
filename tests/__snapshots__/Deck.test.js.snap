// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderBottomColor": "black",
        "borderBottomWidth": 2,
        "paddingBottom": 20,
        "paddingTop": 20,
      },
      undefined,
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    onPress={[Function]}
    style={
      Array [
        Object {
          "fontSize": 30,
          "fontWeight": "bold",
        },
        undefined,
      ]
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    1 card
  </Text>
</View>
`;
